.github/workflows/test_asterinas_nix.yml

```diff
@@ -0,0 +1,163 @@
+  schedule:
+    - cron: '0 22 * * *'
+
+jobs:
+  # Warm-up Nix, so subsequent jobs can reuse the cache
+  nix-cache:
+    if: github.event_name == 'push' || github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: nixbuild/nix-quick-install-action@v28
+
+      - name: Calculate Cache Key
+        id: calc-cache-key
+        run: |
+          key=$(nix derivation show '.#_cache' | jq 'keys[0]' -r)
+          echo "key=$key" >> "$GITHUB_OUTPUT"
+
+      - name: Try Restore Nix Store Cache
+        id: nix-cache-restore
+        uses: nix-community/cache-nix-action@v5.1.0
+        with:
+          primary-key: nix-cache-${{ runner.os }}-${{ steps.calc-cache-key.outputs.key }}
+          purge-prefixes: nix-cache-${{ runner.os }}-
+          skip-restore-on-hit-primary-key: true
+
+      - name: Build Nix Cache
+        if: steps.nix-cache-restore.outputs.hit-primary-key != 'true'
+        run: |
+          nix build .#_cache -L
+          nix-store --add-root ./result --realise
+          nix store gc
+
+  lint:
+    if: github.event_name == 'push' || github.event_name == 'pull_request'
+    needs: nix-cache
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: nixbuild/nix-quick-install-action@v28
+
+      - name: Restore Nix Store Cache
+        uses: nix-community/cache-nix-action@v5.1.0
+        with:
+          primary-key: "nix-cache-dummy"
+          restore-prefixes-first-match: nix-cache-${{ runner.os }}-
+          fail-on: first-match.miss
+          save: false
+
+      - name: Setup Environment using Nix
+        uses: rrbutani/use-nix-shell-action@v1
+        with:
+          extraNixOptions: "-L"
+
+      - name: Check
+        id: check
+        run: make check
+
+  unit-test:
+    if: github.event_name == 'push' || github.event_name == 'pull_request'
+    needs: nix-cache
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: nixbuild/nix-quick-install-action@v28
+
+      - name: Restore Nix Store Cache
+        uses: nix-community/cache-nix-action@v5.1.0
+        with:
+          primary-key: "nix-cache-dummy"
+          restore-prefixes-first-match: nix-cache-${{ runner.os }}-
+          fail-on: first-match.miss
+          save: false
+
+      - name: Setup Environment using Nix
+        uses: rrbutani/use-nix-shell-action@v1
+        with:
+          extraNixOptions: "-L"
+
+      - name: Usermode Unit test
+        id: usermode_unit_test
+        run: make test
+
+      - name: Ktest Unit Test
+        id: ktest_unit_test
+        run: |
+          mkdir -p test/build && cp $PREBUILT_INITRAMFS test/build/initramfs.cpio.gz
+          make ktest
+
+      # TODO: add component check.
+
+  integration-test:
```

Comment:
The three jobs,   lint, unit-test, and integration-test, share the same steps for installing Nix, restoring the cache, and setting up Nix. Is there a way to avoid repeating these steps?

Reply:
In yaml, anchors can be used to reference and use the same data multiple times within a single YAML document, which can avoid repeating these steps. But sadly, GitHub Actions don't support this feature yet, see https://github.com/actions/runner/issues/1182.